Flaky multi processing tests

/kind bug

**What steps did you take and what happened:**
```
=========================== short test summary info ============================
FAILED kserve/test/test_server.py::TestMutiProcessServer::test_rest_server_multiprocess - AssertionError: Not all processes are alive
assert False
 +  where False = all(<generator object TestMutiProcessServer.test_rest_server_multiprocess.<locals>.<genexpr> at 0x7f2e84161f20>)
============ 1 failed, 367 passed, 25 warnings in 161.10s (0:02:41) ============
```


**What did you expect to happen:**


**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Istio Version:
- Knative Version:
- KServe Version:
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

--- a/python/kserve/test/test_server.py
+++ b/python/kserve/test/test_server.py
@@ -1257,8 +1257,16 @@ class TestMutiProcessServer:
         # Start server
         task = asyncio.create_task(server.start())
 
-        # Give some time for processes to start
-        await asyncio.sleep(10)
+        # Wait for processes to start and become ready
+        max_attempts = 30
+        for attempt in range(max_attempts):
+            await asyncio.sleep(1)
+            if len(server._processes) == workers and all(
+                p.is_alive() for p in server._processes
+            ):
+                break
+            if attempt == max_attempts - 1:
+                raise RuntimeError("Server worker processes did not start properly")
 
         try:
             # Check if all worker processes are alive

--- a/python/kserve/test/test_server.py
+++ b/python/kserve/test/test_server.py
@@ -1269,14 +1269,8 @@ class TestMutiProcessServer:
                 raise RuntimeError("Server worker processes did not start properly")
 
         try:
-            # Check if all worker processes are alive
-            assert len(server._processes) == workers
-            assert all(
-                p.is_alive() for p in server._processes
-            ), "Not all processes are alive"
-
             # Constantly poll to check if the port is open
-            for _ in range(10):
+            for _ in range(30):
                 try:
                     resp = httpx.get("http://localhost:8080/")
                     if resp.status_code == 200:

